modules/phase_field/src/materials/GammaAnisoGauss.C

```diff
@@ -0,0 +1,582 @@
+  params.addParam<MaterialPropertyName>("qwg_name", "qwg_name", "w component of quaternion.");
+  params.addParam<MaterialPropertyName>("qxg_name", "qxg_name", "x component of quaternion.");
+  params.addParam<MaterialPropertyName>("qyg_name", "qyg_name", "y component of quaternion.");
+  params.addParam<MaterialPropertyName>("qzg_name", "qzg_name", "z component of quaternion.");
+  params.addRequiredParam<FileName>("Library_file_name",
+                                    "Name of the file containing anisotropic data.");
+  params.addRequiredParam<FileName>("Quaternion_file_name",
+                                    "Name of the file containing quaternion data.");
+  params.addRequiredCoupledVarWithAutoBuild(
+      "v", "var_name_base", "op_num", "Array of coupled variables.");
+  params.addRequiredParam<Real>("sigmaBASE", "Value of sigma base energy in J/m^2.");
+  params.addRequiredParam<Real>("gammaBASE", "Value of gamma at sigmaBASE energy.");
+  params.addRequiredParam<Real>("GgammaBASE", "Value of Ggamma at gammaBASE energy.");
+  params.addRequiredParam<Real>("f0gammaBASE", "Value of f0gamma at gammaBASE energy.");
+  params.addRequiredParam<Real>("lgbBASE_minimum", "Value of minimum grain boundary width in n.m");
+  params.addRequiredParam<Real>("alphaswitch", "Acceptable range of angle from vbalib.");
+  params.addRequiredParam<Real>("betaswitch", "Acceptable range of angle from thetabalib.");
+  params.addRequiredParam<int>("libnum", "Number of misorientations in the library.");
+  params.addRequiredParam<Real>("amplitudeScale", "Amplitude scale of gaussians.");
+  params.addRequiredParam<Real>("sharpness", "Sharpness of gaussians.");
+  params.addRequiredParam<Real>("Gaussian_Tolerance", "Gaussian Tolerance.");
+  params.addRequiredParam<UserObjectName>("grain_tracker", "Graintracker UserObject.");
+  params.addRequiredParam<bool>("ADDGaussian",
+                                "If this is true gaussians add to gammaBASE, else subtract.");
+  params.addParam<Real>("length_scale", 1.0e-9, "Conversion for 1/m to 1/nm.");
+  params.addParam<Real>("time_scale", 1.0e-9, "Time scale in s, where default is ns.");
+  params.addParam<Real>("JtoeV", 6.24150974e18, "Joule to eV conversion.");
+  params.addParam<Real>(
+      "GBMobility",
+      1,
+      "GB mobility input in m^4/(J*s), that overrides the temperature dependent calculation.");
+  params.addParam<Real>("Mob", 0, "Grain boundary mobility prefactor in m^4/(J*s).");
+  params.addParam<Real>("Q", 0, "Grain boundary migration activation energy in eV.");
+  params.addParam<Real>("T", 273, "Temperature in Kelvin.");
+  params.addParam<Real>("kb", 8.617343e-5, "Boltzmann constant in eV/K.");
+  params.addParam<Real>(
+      "BoundaryNormal",
+      0,
+      "If 0, the boundary normal in the simulation reference frame is computed using the "
+      "orientation quaternions, else it is taken directly from the minima library file.");
+
```

Comment:
add parameter groups for readibility of the doco

params.addParamNamesToGroup

Reply:
I ended up removing that. Made a bool.